api/src/main/java/org/mapstruct/tools/gem/GemValue.java

```diff
@@ -1,52 +1,150 @@
 package org.mapstruct.tools.gem;
 
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.AnnotationMirror;
 import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.VariableElement;
 
 /**
  * Class representing a annotation value
  *
  * @param <T> the type represented by this annotation value
  */
-public interface GemValue<T> {
+public class GemValue<T> {
```

Comment:
The risk here is to have a considerable amount of logic in a public package..

Reply:
Also, we might extend in the future by allowing more fields on the annotation for the user to point out his own `GemValue` classes an interface is perhaps a better way to do this than a class.. Or am I over designing this :smile: ?